File: frontend/src/lib/api/trust_analytics.ts
import { apiRequest } from "@/lib/api/request";

export type TrustAnalyticsDateRange = {
  from: string;
  to: string;
};

export type TrustDistribution = {
  high_count: number;
  medium_count: number;
  low_count: number;
  warning_count: number;
  not_found_count: number;
  high_pct: number | null;
  medium_pct: number | null;
  low_pct: number | null;
  warning_pct: number | null;
  not_found_pct: number | null;
};

export type WarningBreakdown = {
  stale_source_count: number;
  conflict_count: number;
  ocr_count: number;
  extraction_count: number;
  processing_count: number;
  evidence_quality_count: number;
  citation_validation_failed_count: number;
};

export type TrustTrendPoint = {
  date: string;
  answer_count: number;
  not_found_count: number;
  not_found_rate: number | null;
  avg_confidence_score: number | null;
  avg_citation_support_score: number | null;
  high_trust_count: number;
  low_trust_count: number;
};

export type LangfuseIntegrationStatus = {
  enabled: boolean;
  traces_linked_count: number;
};

export type TrustAnalyticsResponse = {
  organization_id: string;
  range: TrustAnalyticsDateRange;
  generated_at: string;
  total_answers: number;
  not_found_rate: number | null;
  avg_confidence_score: number | null;
  avg_citation_support_score: number | null;
  avg_verification_support_score: number | null;
  unsupported_claims_removed_total: number;
  conflict_detection_rate: number | null;
  trust_distribution: TrustDistribution;
  warnings: WarningBreakdown;
  daily_trends: TrustTrendPoint[];
  langfuse: LangfuseIntegrationStatus;
  telemetry_missing: boolean;
};

export type TrustAnalyticsQuery = {
  from?: string;
  to?: string;
};

export async function getTrustAnalytics(
  query: TrustAnalyticsQuery = {},
): Promise<TrustAnalyticsResponse> {
  const params = new URLSearchParams();
  if (query.from) params.set("from", query.from);
  if (query.to) params.set("to", query.to);
  const qs = params.toString();
  return apiRequest<TrustAnalyticsResponse>(
    `/admin/trust-analytics${qs ? `?${qs}` : ""}`,
  );
}

File: backend/app/domains/admin/schemas/trust_analytics.py
"""Pydantic schemas for the admin trust analytics API (F317).

Returned by GET /admin/trust-analytics. Aggregates trust metric events
emitted by TrustMetricsService without exposing raw question/answer content.
"""

from __future__ import annotations

from datetime import date, datetime
from typing import Literal

from pydantic import BaseModel, Field


class TrustAnalyticsDateRange(BaseModel):
    from_date: date = Field(alias="from")
    to_date: date = Field(alias="to")

    model_config = {"populate_by_name": True}


class TrustDistribution(BaseModel):
    """Per-level counts and percentage share of answered turns."""

    high_count: int = 0
    medium_count: int = 0
    low_count: int = 0
    warning_count: int = 0
    not_found_count: int = 0
    high_pct: float | None = None
    medium_pct: float | None = None
    low_pct: float | None = None
    warning_pct: float | None = None
    not_found_pct: float | None = None


class WarningBreakdown(BaseModel):
    """Counts of answers that triggered each warning type."""

    stale_source_count: int = 0
    conflict_count: int = 0
    ocr_count: int = 0
    extraction_count: int = 0
    processing_count: int = 0
    evidence_quality_count: int = 0
    citation_validation_failed_count: int = 0


class TrustTrendPoint(BaseModel):
    """Single day data point for trend charts."""

    date: date
    answer_count: int
    not_found_count: int
    not_found_rate: float | None = None
    avg_confidence_score: float | None = None
    avg_citation_support_score: float | None = None
    high_trust_count: int = 0
    low_trust_count: int = 0


class LangfuseIntegrationStatus(BaseModel):
    """Whether Langfuse trace linking is available for trust events."""

    enabled: bool
    traces_linked_count: int = 0


class TrustAnalyticsResponse(BaseModel):
    """Aggregated trust analytics for an org over a date range."""

    organization_id: str
    range: TrustAnalyticsDateRange
    generated_at: datetime

    total_answers: int
    not_found_rate: float | None = None
    avg_confidence_score: float | None = None
    avg_citation_support_score: float | None = None
    avg_verification_support_score: float | None = None
    unsupported_claims_removed_total: int = 0
    conflict_detection_rate: float | None = None

    trust_distribution: TrustDistribution
    warnings: WarningBreakdown
    daily_trends: list[TrustTrendPoint] = Field(default_factory=list)
    langfuse: LangfuseIntegrationStatus

    telemetry_missing: bool = False
